internal/app/api/handler/user/user.go

```diff
@@ -84,7 +101,7 @@ func (h *Handler) GetUserByID(w http.ResponseWriter, r *http.Request) {
 
 	user, err := h.srv.FindUserById(r.Context(), mux.Vars(r)["id"])
 	if err != nil {
-		respond.JSON(w, http.StatusUnauthorized, h.em.InvalidValue.Request)
+		respond.JSON(w, http.StatusInternalServerError, h.em.InvalidValue.Request)
```

Comment:
use if err:= ...; err {} incase no new variable is created
Log error before response